pkg/prometheus_ingester/prometheus_ingester.go

```diff
@@ -153,9 +196,17 @@ func New(initConfig PrometheusIngesterConfig, logger logrus.FieldLogger) (*Prome
 		ingester       = PrometheusIngester{}
 	)
 
+	httpHeaders := map[string]string{}
+	for _, h := range initConfig.HttpHeaders {
+		httpHeaders[h.Name] = h.Value
+	}
+
 	client, err := api.NewClient(api.Config{
-		Address:      initConfig.ApiUrl,
-		RoundTripper: initConfig.RoundTripper,
+		Address: initConfig.ApiUrl,
+		RoundTripper: httpHeadersRoundTripper{
+			rt:      initConfig.RoundTripper,
```

Comment:
I'm not quite sure why RoundTripper is to be stored within config object.

What do you say to drop it and change this as follows:
```suggestion
			rt:      api.DefaultRoundTripper,
```

Reply:
this makes sense, but we use initConfig.RoundTripper to overwrite roundtripper in tests, so there would be more refactoring if we want to drop it here

Reply:
Ok, then. Thanks.